packages/amplify-graphql-model-transformer/src/resources/custom-resource-lambda/custom-resource-handler.ts

```diff
@@ -0,0 +1,440 @@
+  event: AWSCDKAsyncCustomResource.IsCompleteRequest,
+): Promise<AWSCDKAsyncCustomResource.IsCompleteResponse> => {
+  log('got event', event);
+  if (event.RequestType === 'Delete') {
+    // nothing else to do on delete
+    console.log('Delete is finished');
+    return finished;
+  }
+  if (!event.PhysicalResourceId) {
+    throw new Error('PhysicalResourceId not set in call to isComplete');
+  }
+  console.log('Fetching current table state');
+  const describeTableResult = await retry(
+    async () => await ddbClient.describeTable({ TableName: event.PhysicalResourceId! }).promise(),
+    (result) => !!result?.Table,
+  );
+  if (describeTableResult.Table?.TableStatus !== 'ACTIVE') {
+    console.log('Table not active yet');
+    return notFinished;
+  }
+  // table is active, need to check GSI status
+  if (describeTableResult.Table.GlobalSecondaryIndexes?.some((gsi) => gsi.IndexStatus !== 'ACTIVE' || gsi.Backfilling)) {
+    console.log('Some GSI is not active yet');
+    return notFinished;
+  }
+
+  if (event.RequestType === 'Create') {
+    // no additional updates required on create
+    console.log('Create is finished');
+    return finished;
+  }
+
+  // need to check if any more GSI updates are necessary
+  const endState = extractTableInputFromEvent(event);
+  const nextUpdate = getNextGSIUpdate(describeTableResult.Table, endState);
+  log('Computed next update', nextUpdate);
+  if (!nextUpdate) {
+    // current state equals end state so we're done
+    console.log('No additional updates needed. Update finished');
+    return finished;
+  }
+  // don't need to merge gsi updates with other table updates here because those have already been applied in the first update
+  console.log('Initiating table update');
+  await ddbClient.updateTable(nextUpdate).promise();
+  return notFinished;
+  // a response of notFinished in this function will cause the function to be invoked again by the state machine after some time
+};
+
+const finished: AWSCDKAsyncCustomResource.IsCompleteResponse = {
+  IsComplete: true,
+};
+const notFinished: AWSCDKAsyncCustomResource.IsCompleteResponse = {
+  IsComplete: false,
+};
+
+// compares the currentState with the endState to determine a next update step that will get the table closer to the end state
+export const getNextGSIUpdate = (currentState: TableDescription, endState: CustomDDB.Input): UpdateTableInput | undefined => {
+  const endStateGSIs = endState.GlobalSecondaryIndexes || [];
+  const endStateGSINames = endStateGSIs.map((gsi) => gsi.IndexName);
+
+  const currentStateGSIs = currentState.GlobalSecondaryIndexes || [];
+  const currentStateGSINames = currentStateGSIs.map((gsi) => gsi.IndexName);
+
+  // function to identify any GSIs that need to be removed
+  const gsiRequiresReplacementPredicate = (currentGSI: DynamoDB.GlobalSecondaryIndexDescription): boolean => {
+    // check if the index has been removed entirely
+    if (!endStateGSINames.includes(currentGSI.IndexName!)) return true;
```

Comment:
nit: do our linting rules allow if statements to have no-brace conditions?

Reply:
I think so as there are no warning msgs when I run the lint check